### PR TITLE
documentation: add TF 2.4.1 support to sm distributed data parallel docs and other updates

### PR DESCRIPTION
**Pull request by TEChopra1000**
*Description of changes:*
* Adding TF 2.4 support notes into SM distribute model parallel docs

*Testing done:*
`tox -e black-check,flake8,pylint,docstyle,sphinx,doc8 --parallel all`

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.md [hunk 1]
```diff
@@ -17,6 +17,10 @@
 
 - Adds support for `_register_comm_hook` (PyTorch 1.7 only) which will register the callable as a communication hook for DDP. NOTE: Like in DDP, this is an experimental API and subject to change.
 
+### Tensorflow
+
+- Adds support for Tensorflow 2.4
```
Comment (anirudh2290) [inline doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.md:22]: i think this should say 2.4.1 ?

File: doc/api/training/smp_versions/v1.1.0/smd_model_parallel_tensorflow.rst [hunk 2]
```diff
@@ -81,7 +81,7 @@ TensorFlow API
       [...]
       x = tf.constant(1.2)                     # placed in partition 0
       with smp.partition(1):
-          y = tf.add(x, tf.constant(2.3))      # placed in partition 1
+          y = tf.add(x, tf.constant(2.3.1))      # placed in partition 1
```
Comment (anirudh2290) [inline doc/api/training/smp_versions/v1.1.0/smd_model_parallel_tensorflow.rst:84]: I don't think we want to change it here.

**Review by anirudh2290 — APPROVED** [review]

File: doc/api/training/smp_versions/v1.1.0/smd_model_parallel_tensorflow.rst (hunk 2 above)
Comment (TEChopra1000) [reply doc/api/training/smp_versions/v1.1.0/smd_model_parallel_tensorflow.rst]: I removed this update.

File: doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.md (hunk 1 above)
Comment (TEChopra1000) [reply doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.md]: I've added 2.4.1

**Review by anirudh2290 — APPROVED** [review]

**Comment by anirudh2290** [comment]
Thank you for the fixes @TEChopra1000 !

**Review by ChoiByungWook — APPROVED** [review]

**Pull request merged**